primitives/core/hashing/src/lib.rs

```diff
@@ -35,48 +35,51 @@ pub fn blake2_512_into(data: &[u8], dest: &mut [u8; 64]) {
 
 /// Do a Blake2 512-bit hash and return result.
 pub fn blake2_512(data: &[u8]) -> [u8; 64] {
-	let mut r = [0; 64];
-	blake2_512_into(data, &mut r);
-	r
+	let mut h = blake2::Blake2b512::new();
+	h.update(data);
+	h.finalize().into()
 }
 
+type Blake2b256 = blake2::Blake2b<U32>;
+
 /// Do a Blake2 256-bit hash and place result in `dest`.
 pub fn blake2_256_into(data: &[u8], dest: &mut [u8; 32]) {
-	type Blake2b256 = blake2::Blake2b<U32>;
 	dest.copy_from_slice(Blake2b256::digest(data).as_slice());
 }
 
 /// Do a Blake2 256-bit hash and return result.
 pub fn blake2_256(data: &[u8]) -> [u8; 32] {
-	let mut r = [0; 32];
-	blake2_256_into(data, &mut r);
-	r
+	let mut h = Blake2b256::new();
+	h.update(data);
+	h.finalize().into()
 }
 
+type Blake2b128 = blake2::Blake2b<U16>;
+
 /// Do a Blake2 128-bit hash and place result in `dest`.
 pub fn blake2_128_into(data: &[u8], dest: &mut [u8; 16]) {
-	type Blake2b128 = blake2::Blake2b<U16>;
 	dest.copy_from_slice(Blake2b128::digest(data).as_slice());
 }
 
 /// Do a Blake2 128-bit hash and return result.
 pub fn blake2_128(data: &[u8]) -> [u8; 16] {
-	let mut r = [0; 16];
-	blake2_128_into(data, &mut r);
-	r
+	let mut h = Blake2b128::new();
+	h.update(data);
+	h.finalize().into()
 }
 
+type Blake2b64 = blake2::Blake2b<U8>;
+
 /// Do a Blake2 64-bit hash and place result in `dest`.
 pub fn blake2_64_into(data: &[u8], dest: &mut [u8; 8]) {
-	type Blake2b64 = blake2::Blake2b<U8>;
```

Comment:
Are the `*_into(data, dest)` family of functions relevant? 
Looks like are not used anywere in the codebase and is some user requires something like that it can easily construct it

Reply:
Do you means that should I remove these `xx_into` functions?

Reply:
I usually tend to follow [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle.
But with that I'm not saying that should be a dogma.
I have the (maybe bad?!) attitude to remove unused code especially the trivial bloat that can be trivially implemented using a line of code by the user 😃